researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/TaskHelper.java

```diff
@@ -181,16 +181,29 @@ private void cacheSurveyGuidCreatedOn(@NonNull TaskModel taskModel) {
                         cacheSurveyGuidCreatedOn(taskModel);
                         return taskModel;
                     });
-        } else if (!Strings.isNullOrEmpty(task.taskFileName)) {
-            // Load survey from static JSON.
-            taskModelSingle = Single.just(loadTaskModel(context, task));
         } else {
-            // Unsupported. Return null.
-            return Single.just(null);
+            // Does the taskID match one of our local resources?
+            taskModelSingle = Single.fromCallable(() -> {
+                // There doesn't seem to be a way to check for file existence before we attempt to
+                // open it. If the file exists, it'll throw a FileNotFoundException wrapped in a
+                // RuntimeException. Wrap in a try catch.
+                try {
+                    return resourceManager.getTask(task.taskID).create(context);
+                } catch (RuntimeException ex) {
+                    // Return null. This signals to the ActivitiesFragment that we can't load the
```

Comment:
Nit: more idiomatic way could be Single.just(getTask).onErrorReturn(null)

Reply:
That looks much better. I'll go ahead and change that.